### PR TITLE
Add guidance on bad exit codes

### PR DESCRIPTION
**Pull request by mahesh-panchal**
Add guidance on how to deal with tools with unexpected error codes.

### CODE REVIEW

File: src/content/docs/contributing/modules.md [hunk 1]
```diff
@@ -842,6 +842,23 @@ The key words "MUST", "MUST NOT", "SHOULD", etc. are to be interpreted as descri
     """
     ```
 
+11. Occasionally, some tools do not exit with the expected exit code 0 upon successful use of the tool. In these cases one can use
+    the `||` operator to run another useful command when the exit code is not 0 (for example, testing if a file is not size 0).
+
+    ```nextflow
```
Comment (mashehu) [inline src/content/docs/contributing/modules.md:848]:
```suggestion
    ```nextflow {4}
```

File: src/content/docs/contributing/modules.md [hunk 2]
```diff
@@ -842,6 +842,23 @@ The key words "MUST", "MUST NOT", "SHOULD", etc. are to be interpreted as descri
...
+    tool \\
+      --input $input \\
+      --summary ${prefix}.summary \\
+      || test -s ${prefix}.summary
+    """
+    ```
+
+    See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html) for examples of properties of files which could be tested.
+    Alternate suggestions include using `grep -c` to search for a valid string match, or other tool which will appropriately
+    error when expected output is not successfully created.
```
Comment (mashehu) [inline src/content/docs/contributing/modules.md:860]:
```suggestion
    error when the expected output is not successfully created.
```

**Review by mashehu — APPROVED** [review]

File: src/content/docs/contributing/modules.md (hunk 1 above)
Comment (mahesh-panchal) [reply src/content/docs/contributing/modules.md:848]: What does this do?

Comment (mashehu) [reply src/content/docs/contributing/modules.md:848]: highlight a specific line in the codeblock: https://nf-co.re/docs/contributing/website_markdown#highlight-lines

File: src/content/docs/contributing/modules.md [hunk 3]
```diff
@@ -842,6 +842,23 @@ The key words "MUST", "MUST NOT", "SHOULD", etc. are to be interpreted as descri
     """
     ```
 
+11. Occasionally, some tools do not exit with the expected exit code 0 upon successful use of the tool. In these cases one can use
+    the `||` operator to run another useful command when the exit code is not 0 (for example, testing if a file is not size 0).
+
+    ```nextflow {4}
```
Comment (mahesh-panchal) [inline src/content/docs/contributing/modules.md:848]:
```suggestion
    ```nextflow {6}
```

**Pull request merged**